src/watchers/flagcandprof.js

```diff
@@ -0,0 +1,53 @@
+const {loadConfig} = require('../functions');
+
+const config = loadConfig();
+
+class FlagsHandler {
+
+    constructor() {
+        this.logger = require('../connections/logger')('watcher-multisig', config.logger);
+    }
+
+    async action({doc, dac_directory, db}) {
+        const {
+            action: {
+                account,
+                name,
+                data: {
+                    block,
+                    cand,
+                    dac_id,
+                    reason,
+                    reporter
+                }
+            },
+            block_num
+        } = doc;
+        if (account === config.eos.custodianContract && name === 'flagcandprof'){
+            const flagDocument = {
+                block_num,
+                cand,
+                reason,
+                reporter,
+                dac_id,
+                block,
+            };
+
+            const collection = db.collection('flags');
+            collection.insertOne(flagDocument).then(() => {
```

Comment:
@rkamysz, Dallas recently suggested to _"check the unflagging also works to re-enable viewing"_.

Looking at this `insertOne()` snippet I am assuming that the update might not work.
Or it already handled somewhere else?

Reply:
I think it could be useful to keep all the flag and unflag events so the history can be queried easier for auditing. Then I thought the API could fetch the **most recent** flag/unflag event and use that to apply the state to the matching profile. Do you see a major problem with that approach?

Reply:
Nope, I will apply it now

Reply:
done